From ec7b288bbc563e90f36d0021ddc5295c61048fb6 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Mon, 4 Jan 2021 16:56:38 +0800
Subject: [PATCH] usb: dwc3: core: disable device wakeup for rk356x usb host

RK356x SDK will power off logic and grf module in suspend.
So we don't need to enable device wakeup for rk356x usb host
to keep usb controller PD on. With this patch, it can call
rk356x combphy init by usb_phy_roothub_init during resume,
and fix the usb3.0 host function failure after resume.

Change-Id: I40f9a34b5f2aaeede833f9e46153f60b767ea3f0
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/dwc3/core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index fe68a35be3b9..3e0ee5742042 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -1273,7 +1273,10 @@ static int dwc3_core_init_mode(struct dwc3 *dwc)
 		 * keep PD on and run phy_power_on again to avoid
 		 * phy_power_on failed (error -110) in Rockchip platform.
 		 */
-		device_init_wakeup(dev, true);
+		if (!of_machine_is_compatible("rockchip,rk3568") &&
+		    !of_machine_is_compatible("rockchip,rk3566"))
+			device_init_wakeup(dev, true);
+
 		phy_power_on(dwc->usb3_generic_phy);
 
 		dwc3_set_prtcap(dwc, DWC3_GCTL_PRTCAP_HOST);
-- 
2.35.3

